src/table.rs

```diff
@@ -202,6 +211,50 @@ impl Table {
         self.pages[page.0].assert_type::<T>()
     }
 
+    /// Force initialize the page at the given index.
+    ///
+    /// If the page at the provided index was created using `push_uninit_page`, it
+    /// will be initialized using the provided ingredient data.
+    ///
+    /// Otherwise, the page will be allocated.
+    ///
+    /// # Panics
+    ///
+    /// If `page` is out of bounds or the type `T` is incorrect.
+    #[inline]
+    pub(crate) fn force_page<T: Slot>(
+        &mut self,
+        page_idx: PageIndex,
+        ingredient: IngredientIndex,
+        memo_types: &Arc<MemoTableTypes>,
+    ) {
+        let page = self.pages.get_mut(page_idx.0);
+
+        match page {
+            Some(page) => {
+                // Initialize the page if was created using `push_uninit_page`.
+                if page.slot_type_id == TypeId::of::<DummySlot>() {
+                    *page = Page::new::<T>(ingredient, memo_types.clone());
+                }
+
+                // Ensure the page has the correct type.
+                page.assert_type::<T>();
+            }
+
+            None => {
+                // Create dummy pages until we reach the page we want.
+                while self.page_count() < page_idx.as_usize() {
+                    // We make sure not to claim any intermediary pages for ourselves, as they may
+                    // be required by a different ingredient when it is deserialized.
+                    self.push_uninit_page();
+                }
```

Comment:
How expensive is this in terms of memory usage to push those dummy pages? Do they allocate all entries (but empty)?

Reply:
Edit: I just saw below that we don't allocate any data.